### PR TITLE
[BF] Fixes issue #849

### PR DESCRIPTION
**Pull request by dohmatob**
Fixes issue #849.

### CODE REVIEW

File: nilearn/decoding/space_net.py [hunk 1]
```diff
@@ -939,6 +940,9 @@ def predict(self, X):
             Predicted class label per sample.
         """
         # cast X into usual 2D array
+        if not hasattr(self, "masker_"):
+            raise NotFittedError("This %s instance is not fitted yet" % (
```
Comment (banilo) [inline nilearn/decoding/space_net.py]: rather "has not been fitted.."?

File: nilearn/decoding/space_net.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
 from functools import partial
 import numpy as np
 from scipy import stats, ndimage
+from sklearn.exceptions import NotFittedError
```
Comment (AlexandreAbraham) [inline nilearn/decoding/space_net.py]: Is it compliant with the minimal version of sklearn requested? I think this error is something recent.

Comment (dohmatob) [inline nilearn/decoding/space_net.py]: Oops!, it didn't occur to me. I'll just issue a RuntimeError then.

On Sun, Nov 22, 2015 at 3:33 PM, Alexandre Abraham <notifications@github.com
> wrote:

> In nilearn/decoding/space_net.py
> <https://github.com/nilearn/nilearn/pull/851#discussion_r45559572>:
>
> > @@ -18,6 +18,7 @@
> >  from functools import partial
> >  import numpy as np
> >  from scipy import stats, ndimage
> > +from sklearn.exceptions import NotFittedError
>
> Is it compliant with the minimal version of sklearn requested? I think
> this error is something recent.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/851/files#r45559572>.
>



-- 
Elvis Dohmatob,
*sent from my  potato...*
http://dohmatob.blogspot.fr/, https://team.inria.fr/parietal/elvis/,
https://github.com/dohmatob

File: nilearn/decoding/space_net.py (hunk 1 above)
Comment (dohmatob) [inline nilearn/decoding/space_net.py]: This is the phrasing from sklearn.

On Sun, Nov 22, 2015 at 3:32 PM, Danilo Bzdok <notifications@github.com>
wrote:

> In nilearn/decoding/space_net.py
> <https://github.com/nilearn/nilearn/pull/851#discussion_r45559567>:
>
> > @@ -939,6 +940,9 @@ def predict(self, X):
> >              Predicted class label per sample.
> >          """
> >          # cast X into usual 2D array
> > +        if not hasattr(self, "masker_"):
> > +            raise NotFittedError("This %s instance is not fitted yet" % (
>
> rather "has not been fitted.."?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/851/files#r45559567>.
>



-- 
Elvis Dohmatob,
*sent from my  potato...*
http://dohmatob.blogspot.fr/, https://team.inria.fr/parietal/elvis/,
https://github.com/dohmatob

**Comment by AlexandreAbraham** [comment]
I like the idea of NotFittedError and I think that we should support it. Maybe we can create a _NotFittedError that is equal to sklearn NotFittedError if it exists and RuntimeError otherwise? This need to be discussed in another PR. In the meantime, +1 for this one.

**Comment by dohmatob** [comment]
On Sun, Nov 22, 2015 at 3:51 PM, Alexandre Abraham <notifications@github.com
> wrote:

> I like the idea of NotFittedError and I think that we should support it.
> Maybe we can create a _NotFittedError that is equal to sklearn
> NotFittedError if it exists and RuntimeError otherwise?
>


> This need to be discussed in another PR.
>
yep.

> In the meantime, +1 for this one.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nilearn/nilearn/pull/851#issuecomment-158764855>.
>



-- 
Elvis Dohmatob,
*sent from my  potato...*
http://dohmatob.blogspot.fr/, https://team.inria.fr/parietal/elvis/,
https://github.com/dohmatob

**Comment by GaelVaroquaux** [comment]
LGTM. Merging
